pytorch_lightning/trainer/trainer.py

```diff
@@ -1240,7 +1239,10 @@ def _teardown(self):
         self.strategy.post_dispatch(self)
         self.strategy.teardown()
         self._data_connector.teardown()
-        self._active_loop.teardown()
+        loop = self._active_loop
+        # loop should never be `None` here but it can because we don't know the trainer stage with `ddp_spawn`
```

Comment:
makes sense. note, we recover the trainer stage in
`                self.strategy._recover_results_in_main_process(spawn_output, self)
`
L673 in trainer.py